core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -467,7 +467,7 @@ private RelationPlan getInsertPlan(
         plan = planner.addRowFilters(
                 table,
                 plan,
-                failIfPredicateIsNotMeet(metadata, session, PERMISSION_DENIED, AccessDeniedException.PREFIX + "Cannot insert into a table restricted with a row filter"),
+                failIfPredicateIsNotMeet(metadata, session, PERMISSION_DENIED, AccessDeniedException.PREFIX + "Cannot insert row that does not match to a row filter"),
```

Comment:
same for updates or deletes?

Reply:
No. It is not possible currently to UPDATE table with row filter at all. For DELETE there is no way to hit an error as you can only delete visible rows.